### PR TITLE
Use Elixir's mix tasks instead of /bin/ scripts

### LINKED ISSUE #24: Use Elixir's mix tasks instead of /bin/ scripts

### PR DESCRIPTION
**Pull request by slavakisel**
closes #24

### CODE REVIEW

File: lib/mix/tasks/deploy.ex [hunk 1]
```diff
@@ -0,0 +1,42 @@
+defmodule Mix.Tasks.Deploy do
+  use Mix.Task
+
+  def run(["production"]) do
```
Comment (ghost) [inline lib/mix/tasks/deploy.ex]: Please prefer single-line syntax `def func, do: ...`

File: lib/mix/tasks/deploy.ex [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    deploy("production", "production")
+  end
+  def run(_) do
+    deploy("staging", "master")
+  end
+
+  defp deploy(dest, branch) do
+    IO.puts("Deploying #{dest} to #{branch}")
```
Comment (ghost) [inline lib/mix/tasks/deploy.ex]: Should we use logger instead of IO?

File: lib/mix/tasks/deploy.ex [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    build(dest, branch) |> release(dest) |> start(dest)
```
Comment (ghost) [inline lib/mix/tasks/deploy.ex]: Pipelines should start from single argument:
```
dest |> build(branch) |> release(dest) |> start(dest)
```

File: lib/mix/tasks/deploy.ex [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  defp build(dest, branch) do
+    cmd("mix edeliver build release #{dest} --branch=#{branch} --verbose")
+  end
+
+  defp release(prev_status, dest) do
```
Comment (ghost) [inline lib/mix/tasks/deploy.ex]: You can use guard clause here for prev_status

File: lib/mix/tasks/deploy.ex [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    if prev_status == 0 do
+      cmd("mix edeliver deploy release #{dest} --verbose")
+    else
+      prev_status
+    end
+  end
+
+  defp start(prev_status, dest) do
```
Comment (ghost) [inline lib/mix/tasks/deploy.ex]: The same guard clause can be used here

File: lib/mix/tasks/run_espec.ex [hunk 6]
```diff
@@ -0,0 +1,7 @@
+defmodule Mix.Tasks.RunEspec do
+  use Mix.Task
+
+  def run(_) do
+    Mix.Shell.cmd("MIX_ENV='test' mix espec --format=doc", fn(output) -> IO.write(output) end)
```
Comment (ghost) [inline lib/mix/tasks/run_espec.ex]: "MIX_ENV='test' mix espec --format=doc" |> Mix.Shell.cmd(&IO.write/1)

File: mix.exs [hunk 7]
```diff
@@ -62,4 +63,20 @@ defmodule ElixirBase.Mixfile do
      {:faker, "~> 0.5", only: :test,}
    ]
   end
+
+  defp aliases do
+    [
+      build: "release",
+      server: "run",
+      quality: ["dialyzer", "credo --strict"],
+      ci: ["run_espec", "quality"],
```
Comment (ghost) [inline mix.exs]: ~w(run_espec quality)

File: lib/mix/tasks/run_espec.ex [hunk 8]
```diff
@@ -0,0 +1,7 @@
+defmodule Mix.Tasks.RunEspec do
```
Comment (ghost) [inline lib/mix/tasks/run_espec.ex:1]: Why not to name it just Espec?

**Review by ghost — CHANGES_REQUESTED** [review]

File: lib/mix/tasks/run_espec.ex (hunk 8 above)
Comment (slavakisel) [reply lib/mix/tasks/run_espec.ex:1]: Didn't want name confusion because there is already existent `espec` task in test environment

Comment (ghost) [reply lib/mix/tasks/run_espec.ex:1]: OK

**Review by ghost — APPROVED** [review]
Nice :shipit:

**Pull request merged**